front/lib/api/assistant/generation.ts

```diff
@@ -451,6 +453,15 @@ export async function constructPromptMultiActions(
     additionalInstructions += `${retrievalMetaPromptMutiActions()}\n`;
     hasAdditionalInstructions = true;
   }
+
+  const hasWebsearch = configuration.actions.some((action) =>
+    isWebsearchConfiguration(action)
+  );
+  if (hasWebsearch) {
+    additionalInstructions += `${websearchMetaPrompt()}\n`;
```

Comment:
I thiiink we want to actually share the metaPrompt for websearch and retrieval as they are ~90% the same and repeating it is weird in case the two actions are available?

Reply:
That was my initial try and I have far less good results than with this one. My guess is that since we provide the url of the page on the content it just does not want to respect the cite. 

But I have not tried to use this one for both.